solutions/cses-1641.mdx

```diff
@@ -51,3 +56,77 @@ int main(){
 	cout << "IMPOSSIBLE" << endl;
 }
 ```
+	
+</CPPSection>
+	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class SumOfThreeValues {
+	static class comparator implements Comparator<Group> {
+		public int compare (Group one, Group two) {
+			return Integer.compare(one.value, two.value);
+		}
+	}
+
+	static class Group {
+		int value;
+		int index;
+	}
+
+	public static void main (String[] args) {
+		Kattio io = new Kattio();
+		int n = io.nextInt();
+		int x = io.nextInt();
+		Group[] data = new Group[n];
+
+		for (int i = 1; i <= n; i++) {
+			// store the input in a class that has a value and an index
+			Group group = new Group();
+			group.value = io.nextInt();
+			group.index = i;
+
+			data[i - 1] = group;
+		}
+
+		Arrays.sort(data, new comparator());  // make the values in increasing order
+
+		for (int i = 0; i < n; i++) {
+			int left = 0;
+			int right = n - 1;
+
+			while (left != right) {
+				int target = x - data[i].value;
+				// if the left and the right are not i, their values sum up to the target
+				if (left != i && right != i && data[left].value + data[right].value == target) {
+					// we have an answer, output it
+					io.println(data[i].index + " " + data[left].index + " " + data[right].index);
```

Comment:
I'm p sure these lines exceed 80 characters

Reply:
barely exceeds on a few lines so I think it should be fine

Reply:
![image](https://user-images.githubusercontent.com/55369003/129505592-9949dc9e-01c5-4ecf-9c1b-37359f1d5c11.png)
"barely"